sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -2101,3 +2101,53 @@ object AsOfJoin {
     }
   }
 }
+
+
+/**
+ * A logical plan for summary.
+ */
+case class UnresolvedSummary(
+    child: LogicalPlan,
+    statistics: Seq[String]) extends UnaryNode {
+
+  private lazy val supported =
+    Set("count", "count_distinct", "approx_count_distinct", "mean", "stddev", "min", "max")
+
+  {
+    // TODO: throw AnalysisException instead
+    require(statistics.nonEmpty)
+    val percentiles = statistics.filter(p => p.endsWith("%")).map { p =>
+      try {
+        p.stripSuffix("%").toDouble / 100.0
+      } catch {
+        case e: NumberFormatException =>
+          throw QueryExecutionErrors.cannotParseStatisticAsPercentileError(p, e)
+      }
+    }
+    require(percentiles.forall(p => p >= 0 && p <= 1), "Percentiles must be in the range [0, 1]")
+
+    statistics.foreach {
+      case s if supported.contains(s) =>
+      case p if p.endsWith("%") =>
+      case s => throw QueryExecutionErrors.statisticNotRecognizedError(s)
+    }
+  }
+
+  override protected def stringArgs: Iterator[Any] = super.stringArgs.take(5)
+
+  override lazy val resolved = false // Summary will be replaced after being resolved.
+
+  final override val nodePatterns: Seq[TreePattern] = Seq(SUMMARY)
+
+  override def output: Seq[Attribute] = {
+    AttributeReference("summary", StringType)() +:
+      child.output.flatMap { attr =>
+        if (attr.dataType.isInstanceOf[NumericType] || attr.dataType.isInstanceOf[StringType]) {
+          Some(AttributeReference(attr.name, StringType)())
+        } else None
```

Comment:
Why we need `None` to  be `placeholder` than just skipping the attribute in the `Summary` outputs?

Reply:
this is a `flatMap`, and `None` will be ignored
currently datatypes other than `NumericType ` and `StringType ` will be discarded